pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -373,6 +373,31 @@ static char *PyTimeToJSON(JSOBJ _obj, JSONTypeContext *tc, size_t *outLen) {
   return outValue;
 }
 
+static char *PyDecimalToUTF8Callback(JSOBJ _obj, JSONTypeContext *tc,
+                                     size_t *len) {
+  PyObject *obj = (PyObject *)_obj;
+  PyObject *str = PyObject_Str(obj);
+  if (str == NULL) {
+    *len = 0;
+    if (!PyErr_Occurred()) {
+      PyErr_SetString(PyExc_ValueError, "Failed to convert decimal");
+    }
+    ((JSONObjectEncoder *)tc->encoder)->errorMsg = "";
+    return NULL;
+  }
+  if (PyUnicode_Check(str)) {
```

Comment:
I think this check is also unnecessary - `PyObject_Str` is guaranteed to return a unicode object, so no need to check again